### ISSUE TITLE
kubectl wait --all with --all-namespaces doesn't find resources

### ISSUE THREAD

**Issue by nbwest** [kubernetes/kubernetes#79121]
Running the command
kubectl wait --for=condition=Ready pods --all --all-namespaces
results in 
error: no matching resources found**What happened**:
However, running 
kubectl wait --for=condition=Ready pods --all -n kube-system
provides expected output with pods in namespace being enumerated.

**What you expected to happen**:
All pods in all namespaces would be checked against the Ready condition

**How to reproduce it (as minimally and precisely as possible)**:
Run the following against a cluster containing pods.
kubectl wait --for=condition=Ready pods --all --all-namespaces

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:45:25Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration:
AWS

- OS (e.g: `cat /etc/os-release`):
NAME="Ubuntu"
VERSION="18.04.2 LTS (Bionic Beaver)"

- Kernel (e.g. `uname -a`):
Linux ubuntu 4.18.0-18-generic #19~18.04.1-Ubuntu SMP Fri Apr 5 10:22:13 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by nbwest** [comment]
@kubernetes/sig-apps-bugs

**Comment by pswica** [comment]
I could not replicate. I tested on minikube
```
kubectl wait --for=condition=Ready pods --all --all-namespaces
pod/annotationtest-7c65dbd89c-xnvxg condition met
pod/ctest-6c9b69f46c-qnkrw condition met
```
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T16:23:09Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T16:14:56Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by nbwest** [comment]
The cluster I see this behaviour on was stood up using kops.
Using another cluster stood up manually I get some results, but it then appears to hang timing out on a large number of completed pods.
When I can get some time I'll try and reproduce it on a manually built cluster

**Comment by DavidWittman** [comment]
I ran into this issue when executing `kubectl` from within a pod. It seems to *always* use the namespace defined in `/var/run/secrets/kubernetes.io/serviceaccount/namespace` even if `-A` is supplied on the command-line.

**Comment by DavidWittman** [comment]
Seems fixed in kubectl v1.18.3, but probably sooner than that release as well.
